frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-metrics/PipelineMetrics.tsx

```diff
@@ -21,12 +21,16 @@ import './PipelineMetrics.scss';
 
 interface PipelineMeticsProps {
   obj: PipelineKind;
+  customData: {
+    queryPrefix: string;
+  };
 }
 
-const PipelineMetrics: React.FC<PipelineMeticsProps> = ({ obj }) => {
+const PipelineMetrics: React.FC<PipelineMeticsProps> = ({ obj, customData }) => {
```

Comment:
The types will probably all be the same, perhaps we even go so far as going to:
```suggestion
const PipelineMetrics: React.FC<PipelineDetailsTabProps> = ({ obj, customData }) => {
```

And go with the common type:
```
type PipelineDetailsTabProps = {
  obj: PipelineKind;
  customData: {
    templateNames: RouteTemplate[];
    queryPrefix: string;
  };
};
```

Reply:
Could you also get all the `Pipeline` details page tabs to be this type too, even if you didn't touch them.

Reply:
Having a common type for all the pipeline details tabs.